Source/Dafny/Compiler-Csharp.cs

```diff
@@ -176,14 +176,14 @@ public CsharpCompiler(ErrorReporter reporter)
       //   }
 
       var cw =
-        CreateClass(IdName(iter), iter.TypeArgs, wr) as CsharpCompiler.ClassWriter;
+        CreateClass(iter.Module.CompileName, IdName(iter), iter.TypeArgs, wr) as CsharpCompiler.ClassWriter;
```

Comment:
Here and throughout, shouldn't this be `IdName(iter.Module)` or `IdProtect(iter.Module.CompileName)` in order to replace identifier characters that are not supported in the target language?

Reply:
I'm actually rather hazy on the distinction between `IdName`, `IdProtect`, and `x.CompileName`.  The latter seems to sometimes be computed from one of the previous two, and sometimes passed to them.  If there's a general rule, I'm happy to apply it throughout.

Reply:
I wrapped all of the new `x.Module.CompileName` in `IdProtect`.